src/icp_v2.cc

```diff
@@ -604,24 +604,6 @@ icpHandleIcpV2(int fd, Ip::Address &from, char *buf, int len)
     }
 }
 
-#ifdef ICP_PKT_DUMP
-static void
-icpPktDump(icp_common_t * pkt)
-{
-    Ip::Address a;
-
-    debugs(12, 9, "opcode:     " << std::setw(3) << pkt->opcode  << " " << icp_opcode_str[pkt->opcode]);
-    debugs(12, 9, "version: "<< std::left << std::setw(8) << pkt->version);
-    debugs(12, 9, "length:  "<< std::left << std::setw(8) << ntohs(pkt->length));
-    debugs(12, 9, "reqnum:  "<< std::left << std::setw(8) << ntohl(pkt->reqnum));
-    debugs(12, 9, "flags:   "<< std::left << std::hex << std::setw(8) << ntohl(pkt->flags));
-    a = (struct in_addr)pkt->shostid;
```

Comment:
I agree with both of you: This code should be removed (but _not_ because some external tool can offer similar debugging in some circumstances) and, ideally, new always-compiled high-quality debugging code should be added. Those two changes do not have to happen in a _single_ PR because the being-removed code does not compile if enabled (and was probably dead/unused for a very long time):

```
icp_v2.cc:618:30: error: no matching function for call to ‘in_addr::in_addr(uint32_t&)’
  618 |     a = (struct in_addr)pkt->shostid;
```

Thus, I am approving this PR (with adjusted PR title/description).